src/components/Dashboard/DashboardSettingsModal.tsx

```diff
@@ -0,0 +1,202 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { cloneDeep, toNumber } from 'lodash';
+
+import Button from '@/components/Input/Button';
+import Checkbox from '@/components/Input/Checkbox';
+import InputSmall from '@/components/Input/InputSmall';
+import ModalPanel from '@/components/Panels/ModalPanel';
+import { useGetSettingsQuery, usePatchSettingsMutation } from '@/core/rtkQuery/splitV3Api/settingsApi';
+import { setLayoutEditMode } from '@/core/slices/mainpage';
+import { initialSettings } from '@/pages/settings/SettingsPage';
+
+type Props = {
+  onClose: () => void;
+  show: boolean;
+};
+
+const Title = ({ onClose }: { onClose: () => void }) => {
+  const dispatch = useDispatch();
+
+  return (
+    <div className="flex items-center justify-between text-xl font-semibold">
+      Display Settings
+      <Button
+        onClick={() => {
+          dispatch(setLayoutEditMode(true));
+          onClose();
+        }}
+        buttonType="primary"
+        className="px-2 py-1 text-sm"
+      >
+        Enable Edit Mode
+      </Button>
+    </div>
+  );
+};
+
+const DashboardSettingsModal = ({ onClose, show }: Props) => {
+  const dispatch = useDispatch();
+
+  const settingsQuery = useGetSettingsQuery();
+  const settings = useMemo(() => settingsQuery?.data ?? initialSettings, [settingsQuery]);
+  const [patchSettings] = usePatchSettingsMutation();
+
+  const [newSettings, setNewSettings] = useState(initialSettings);
+
+  useEffect(() => {
+    setNewSettings(settings);
+  }, [dispatch, settings]);
+
+  const {
+    combineContinueWatching,
+    hideCollectionBreakdown,
+    hideImportFolders,
+    hideQueueProcessor,
+    hideR18Content,
+    hideRecentlyImported,
+    hideShokoNews,
+    recentlyImportedEpisodesCount,
+    recentlyImportedSeriesCount,
+    shokoNewsPostsCount,
+  } = newSettings.WebUI_Settings.dashboard;
+
+  const updateSetting = (key: keyof typeof settings.WebUI_Settings.dashboard, value: boolean | number) => {
+    const tempSettings = cloneDeep(newSettings);
+    if (
+      key === 'recentlyImportedEpisodesCount' || key === 'recentlyImportedSeriesCount' || key === 'shokoNewsPostsCount'
+    ) {
+      tempSettings.WebUI_Settings.dashboard[key] = value as number;
+    } else {
+      tempSettings.WebUI_Settings.dashboard[key] = value as boolean;
+    }
+    setNewSettings(tempSettings);
+  };
+
+  const handleSave = async () => {
+    try {
+      await patchSettings({ oldSettings: settings, newSettings }).unwrap();
+      onClose();
+    } catch (error) { /* empty */ }
```

Comment:
We should _at least_ log the error, and maybe handle the error. The user might start wondering _why_ it won't save properly without any indicators.

Reply:
The error is shown in the toast notification already if it fails